includes/class-regeneratethumbnails-regenerator.php

```diff
@@ -313,7 +322,7 @@ public function filter_image_sizes_to_only_missing_thumbnails( $sizes, $fullsize
 			return $sizes;
 		}
 
-		$metadata = wp_get_attachment_metadata( $this->attachment->ID );
+		$metadata = $this->old_metadata;
```

Comment:
Probably a quite silly thing but, WDYT about resetting the `old_metadata` property after the `remove_filter` call for this filter? 

The reason I'm asking this is that, while the property itself is private, it keeps the internal state of the attachment being processed. Hence, if a third party for some reason decides to call this method (which it shouldn't, but oh well, people are people.. and technically better safe than sorry), then it will find unexpected results because of the internal state saved in `old_metadata`.

Based on this, my suggestion would be to just reset `old_metadata` to an empty array, and then check here if this variable is `empty`, and if it is, we just return early with the `$sizes` variable, without doing anything else.

Reply:
Hey Fernando, initially I thought it made a lot of sense. But when implementing the changes I realised that this class i already quite dependent on its internal status (like `attachment->ID`, `fullsizepath` and so on).

So for the moment I am leaving it as it is, so we don't add more complexity to the logic.